### PR TITLE
[Refactor](auth)(step-2) Add AccessController to support customized authorization

### PR DESCRIPTION
**Pull request by morningman**
# Proposed changes

Issue Number: #16762

## Problem summary

1. Support specifying AccessControllerFactory when creating catalog

    ```
    create catalog hive properties(
    ...
    "access_controller.class" = "org.apache.doris.mysql.privilege.RangerAccessControllerFactory",
    "access_controller.properties.prop1" = "xxx",
    "access_controller.properties.prop2" = "yyy",
    ...
    )
    ```

    So that user can specified their own access controller, such as RangerAccessController

2. Add interface to check column level privilege

    A new method of CatalogAccessController: `checkColsPriv()`,
    for checking column level privileges.

    TODO:
    Support grant column level privileges statements in Doris

3. Add TestExternalCatalog/Database/Table/ScanNode

    These classes are used for FE unit test. In unit test you can
    
    ```
    create catalog test1 properties(
        "type" = "test"
        "catalog_provider.class" = "org.apache.doris.datasource.ColumnPrivTest$MockedCatalogProvider"
        "access_controller.class" = "org.apache.doris.mysql.privilege.TestAccessControllerFactory",
        "access_controller.properties.key1" = "val1",
        "access_controller.properties.key2" = "val2"
    );
    ```

    To create a test catalog, and specify `catalog_provider` to mock database/table/schema metadata

4. Set roles in `current user identity` in connection context

    The roles can be used for authorization in access controller.

## Checklist(Required)

* [ ] Does it affect the original behavior
* [x] Has unit tests been added
* [ ] Has document been added or modified
* [ ] Does it need to update dependencies
* [ ] Is this PR support rollback (If NO, please explain WHY)

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: fe/fe-core/src/main/java/org/apache/doris/catalog/external/TestExternalTable.java [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+import org.apache.doris.thrift.TTableDescriptor;
+import org.apache.doris.thrift.TTableType;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+
+import java.util.List;
+
+/**
+ * Elasticsearch external table.
```
Comment (Yulei-Yang) [inline fe/fe-core/src/main/java/org/apache/doris/catalog/external/TestExternalTable.java:31]: typo, es?

File: fe/fe-core/src/main/java/org/apache/doris/mysql/privilege/AccessControllerManager.java [hunk 2]
```diff
@@ -51,8 +57,21 @@ private CatalogAccessController getAccessControllerOrDefault(String ctl) {
...
+            ctlToCtlAccessController.put(ctl, accessController);
+            LOG.info("create access controller {} for catalog {}", ctl, acFactoryClassName);
+        } catch (ClassNotFoundException e) {
+            throw new RuntimeException(e);
+        } catch (InstantiationException e) {
+            throw new RuntimeException(e);
+        } catch (IllegalAccessException e) {
+            throw new RuntimeException(e);
+        }
     }
```
Comment (Yulei-Yang) [inline fe/fe-core/src/main/java/org/apache/doris/mysql/privilege/AccessControllerManager.java:76]: need a deleteAccessController method

Comment (morningman) [reply fe/fe-core/src/main/java/org/apache/doris/mysql/privilege/AccessControllerManager.java:80]: fixed. the access controller will be removed when catalog is dropped

**Review by yiguolei — APPROVED** [review]
LGTM

**Pull request merged**